docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,426 @@
+
+#### Revalidating
+
+React's `cache` function is used specifically for memoizing the return values of functions. Since the memoization is not shared across requests and only applies during the rendering of the React tree, there is no need for revalidation.
+
+#### Opting out
+
+To opt out of memoization in `fetch` requests, you can pass an `AbortController` `signal` to the request.
+
+```js
+const { signal } = new AbortController()
+fetch(url, { signal })
+```
+
+Since you manually opt into using the `cache` function, there's no way to opt out.
+
+### Data Cache
+
+Next.js has a built-in Data Cache that stores the result of data requests across user requests and deployments.
+
+By default, data requests that use `fetch` are automatically cached. This is possible because Next.js extends the native `fetch` API to allow each request on the server to set its own persistent caching semantics.
+
+You can use the [`cache`](#fetchoptionscache) and [`next.revalidate`](#fetchoptionsnextrevalidate) options of `fetch` to configure the caching behavior. In the browser, `cache` indicates how a fetch request will interact with the browser's HTTP cache, and with Next.js, the `cache` option also indicates how a server-side fetch request will interact with the framework's Data Cache.
+
+When fetching data on the server, a `fetch` request first checks the Data Cache for a cached response. If a cached response is found, it is returned immediately. If not, the request is made to the data source, and the response is cached.
+
+> **Static vs. Dynamic Data Fetching**:
+>
+> You may see the terms **Static Data Fetching** and **Dynamic Data Fetching** being used. Static Data Fetching refers to cached data requests (default), whereas Dynamic Data Fetching refers to data that is fetched on every request, at runtime.
+
+#### Duration
+
+The Data Cache is persistent across user requests and deployments unless you revalidate or opt-out.
+
+#### Revalidating
+
+Revalidation is the process of purging the Data Cache and re-fetching the latest data. This is useful when your data changes and you want to ensure you show the latest information.
+
+Cached data can be revalidated in two ways:
+
+- **Time-based revalidation**: Automatically revalidate data after a certain amount of time has passed. This is useful for data that changes infrequently and freshness is not as critical. Time-based revalidation can be configured with the [Route Segment Config options](#segment-config-options) or per individual [`fetch` request](#fetchoptionsnextrevalidate).
+- **On-demand revalidation**: Manually revalidate data based on an event (e.g. form submission). On-demand revalidation can use a tag-based or path-based approach to revalidate groups of data at once. This is useful for when you want to ensure the latest data is always shown. On-demand revalidation can be configured with the [`revalidateTag`](#fetchoptionsnexttag-and-revalidatetag) and [`revalidatePath`](#revalidatepath) APIs.
```

Comment:
On-demand revalidation removes the entry from the cache, the first time the request is executed again this will cause a cache miss and thus populate the Data Cache with the new result. So there is no stale-while-revalidate behavior in this case, this is to ensure consistency.

Reply:
ty, added section on how on-demand revalidation works in the data fetching page.